src/Controller/Pia/ProcessingController.php

```diff
@@ -73,4 +74,145 @@ public function listAction(Request $request)
+     *     )
+     * )
+     *
+     * @Security("is_granted('CAN_CREATE_PROCESSING')")
+     *
+     * @return array
+     */
+    public function createAction(Request $request)
+    {
+        
+        $folder = $this->getResource($request->get('folder_id', -1), Folder::class);
+        $entity = $this->get('jms_serializer')->deserialize($request->getContent(), $this->getEntityClass(), 'json');
+
+        $processing = $this->processingService->createProcessing(
+            $entity->getName(),
+            $folder,
+            $entity->getAuthor(),
+            $entity->getControllers()
+        );
+
+        $this->persist($processing);
+
+        return $this->view($processing, Response::HTTP_OK);
+    }
+
+
+    /**
+     * Updates a processing.
+     *
+     * @Swg\Tag(name="Processing")
+     *
+     * @Swg\Response(
+     *     response=200,
+     *     description="Returns the updated Processing",
+     *     @Swg\Schema(
+     *         type="object",
+     *         ref=@Nelmio\Model(type=Processing::class, groups={"Default"})
+     *     )
+     * )
+     *
+     * @FOSRest\Put("/processings/{id}", requirements={"id"="\d+"})
+     *
+     * @Security("is_granted('CAN_EDIT_PROCESSING')")
+     *
+     * @return array
+     */
+    public function updateAction(Request $request, $id)
+    {
+        $processing = $this->getResource($id);
+        $this->canAccessResourceOr403($processing);
+
+        $updatableAttributes = [
+            'folder'            => Folder::class,
+            'name'              => RequestDataHandler::TYPE_STRING,
+            'author'            => RequestDataHandler::TYPE_STRING,
+            'description'       => RequestDataHandler::TYPE_STRING,
+            'processors'        => RequestDataHandler::TYPE_STRING,
+            'controllers'       => RequestDataHandler::TYPE_STRING,
+            'non_eu_transfer'   => RequestDataHandler::TYPE_STRING,
+            'life_cycle'        => RequestDataHandler::TYPE_STRING,
+            'storage'           => RequestDataHandler::TYPE_STRING,
+            'standards'         => RequestDataHandler::TYPE_STRING,
+            'status'            => RequestDataHandler::TYPE_INT,
+        ];
+
+        $this->mergeFromRequest($processing, $updatableAttributes, $request);
+
+        $this->update($processing);
+
+        return $this->view($processing, Response::HTTP_OK);
+    }
+
+    /**
+     * Deletes a processing.
+     *
+     * @Swg\Tag(name="Processing")
+     *
+     * @Swg\Response(
+     *     response=200,
+     *     description="Empty content"
+     * )
+     *
+     * @FOSRest\Delete("/processings/{id}", requirements={"id"="\d+"})
+     *
+     * @Security("is_granted('CAN_DELETE_PROCESSING')")
+     *
+     * @return array
+     */
+    public function deleteAction(Request $request, $id)
+    {
+        $processing = $this->getResource($id);
+        $this->canAccessResourceOr403($processing);
+
+        if (count($processing->getPias())) {
```

Comment:
Could you add the explicit check `if (count($processing->getPias() > 0)) {` ?

Reply:
Absolutely !